### ISSUE TITLE
[Blazor] debugging startup takes 24 seconds while without it loads instantly

### ISSUE THREAD

**Issue by xperiandri** [dotnet/aspnetcore#49244]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

After I set up my enterprise-grade solution, debug startup time increased from 5 seconds which is pretty nice to 24-28 seconds which is not acceptable

### Expected Behavior

Debugging startup time takes no more than 10 seconds

### Steps To Reproduce

1. `dotnet new -i Bolero.Templates`
2. `dotnet new bolero-app -o MyBoleroApp`
3. Add `"inspectUri": "{wsProtocol}://{url.hostname}:{url.port}/_framework/debug/ws-proxy?browser={browserInspectUri}",` to `launchSettings.json` as per https://github.com/fsbolero/Bolero/issues/307#issuecomment-1593158277
4. Run in debug mode and see that app loads in 5 seconds

5. unpack [App.zip](https://github.com/dotnet/aspnetcore/files/11968285/App.zip) and run in debug mode `Marketplace.Server`
6. See that it loads very slow!

### .NET Version

7.0.102

### Anything else?

Visual Studio 2022 17.7.0 Preview 2.0

**Comment by rogihee** [comment]
Yes, with me on 17.6.4 it takes up to 12 seconds to download the browserLink:
![image](https://github.com/dotnet/aspnetcore/assets/4984486/680efc38-f275-49b1-8c3c-51eb6600aa1d)

Not sure if it's the same issue but it's quite a wait to start every time with the debugger!

**Comment by xperiandri** [comment]
Looks like it was fixed in 8.0
